### PR TITLE
Simple cleanup in XDocument  classes

### PR DESCRIPTION
**Pull request by Maxwe11**

### CODE REVIEW

**Comment by krwq** [comment]
Looks good :+1:

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 1]
```diff
@@ -1359,12 +1338,10 @@ bool IsDuplicateNamespaceAttributeInner(XAttribute candidateAttribute)
                                 // And it's for the same namespace URI as well - so ours is a duplicate
                                 return true;
                             }
-                            else
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 2]
```diff
@@ -1312,11 +1293,9 @@ bool IsDuplicateNamespaceAttribute(XAttribute candidateAttribute)
             {
                 return false;
             }
-            else
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 3]
```diff
@@ -678,12 +661,9 @@ public override bool MoveToAttribute(string localName, string namespaceName)
                                 // If it's a duplicate namespace attribute just act as if it doesn't exist
                                 return false;
                             }
-                            else
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: Leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 4]
```diff
@@ -631,12 +617,9 @@ public override bool MoveToAttribute(string name)
...
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 5]
```diff
@@ -541,10 +530,7 @@ public override string GetAttribute(string localName, string namespaceName)
                             {
                                 return null;
                             }
-                            else
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 6]
```diff
@@ -486,10 +478,7 @@ public override string GetAttribute(string name)
...
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 7]
```diff
@@ -436,8 +430,6 @@ public override XmlSpace XmlSpace
                                     return XmlSpace.Preserve;
                                 case "default":
                                     return XmlSpace.Default;
-                                default:
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave the default.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs [hunk 8]
```diff
@@ -139,15 +139,9 @@ public override bool HasAttributes
                     {
                         return GetFirstNonDuplicateNamespaceAttribute(e.lastAttr.next) != null;
                     }
-                    else
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XNodeReader.cs]: We should leave these else statements.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XName.cs [hunk 9]
```diff
@@ -78,10 +78,7 @@ public static XName Get(string expandedName)
                 if (i <= 1 || i == expandedName.Length - 1) throw new ArgumentException(SR.Format(SR.Argument_InvalidExpandedName, expandedName));
                 return XNamespace.Get(expandedName, 1, i - 1).GetName(expandedName, i + 1, expandedName.Length - i - 1);
             }
-            else
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XName.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XLinq.cs [hunk 10]
```diff
@@ -428,7 +430,7 @@ public string GetPrefixOfNamespace(XNamespace ns, bool allowDefaultNamespace)
                                 _rover = d;
                                 return d.prefix;
                             }
-                            else if (d.prefix.Length > 0)
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XLinq.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XContainer.cs [hunk 11]
```diff
@@ -803,12 +805,12 @@ IEnumerable<XElement> GetElements(XName name)
...
         {
-            string s;
-            if (value is string)
+            string s = value as string;
+            if (s != null)
             {
-                s = (string)value;
+                return s;
             }
-            else if (value is double)
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XContainer.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XAttribute.cs [hunk 12]
```diff
@@ -718,7 +718,7 @@ static void ValidateAttribute(XName name, string value)
                     // be declared by the 'xml' prefix namespace declaration.
                     throw new ArgumentException(SR.Argument_NamespaceDeclarationXml);
                 }
-                else if (value == XNamespace.xmlnsPrefixNamespace)
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XAttribute.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XAttribute.cs [hunk 13]
```diff
@@ -703,7 +703,7 @@ static void ValidateAttribute(XName name, string value)
                         // prefix namespace declaration. 
                         throw new ArgumentException(SR.Argument_NamespaceDeclarationXml);
                     }
-                    else if (localName == "xmlns")
+                    if (localName == "xmlns")
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XAttribute.cs]: We should leave the else.

File: src/System.Xml.XDocument/src/System/Xml/Linq/XAttribute.cs [hunk 14]
```diff
@@ -682,7 +682,7 @@ static void ValidateAttribute(XName name, string value)
                     // the default namespace declaration
                     throw new ArgumentException(SR.Format(SR.Argument_NamespaceDeclarationPrefixed, name.LocalName));
                 }
-                else if (value == XNamespace.xmlPrefixNamespace)
```
Comment (weshaggard) [inline src/System.Xml.XDocument/src/System/Xml/Linq/XAttribute.cs]: We should leave the else.

**Comment by weshaggard** [comment]
@Maxwe11 I would like to thank you for your interest and contributions and I think a lot of them are valuable. However there are also a lot of changes which I think should be reverted because they don't add enough value in comparison to the potential issues they might cause. 

In particular there is one type of change that I commented on in #502 about the else statement removals. At this point in time we still have a internal version of this source code as well and we need to keep changes in sync between them. In our experience those type of changes tend to introduce subtle bugs and issues when people make changes or we merge changes together. So we want to avoid making changes like these that don't have enough value to out weight the risk of introducing bugs. 

Another thing is that we have an formatting tool that automatically does a lot of code clean-up and we have issue #496 which is to run it on the Xml code. We also have a policy on our coding guidelines to avoid submitting PRs for fixing [Style](https://github.com/dotnet/corefx/wiki/Contributing#style) issues for things that we can automate with our tooling. However there are still some changes that are valuable and we cannot or do not currently automate, such as adding readonly to fields and eliminating double casts. 

With that said I am going to revert this PR and I do want to apologize and thank you again for your interest and contributions. 

I would encourage you to submit a new PR that does some of the clean-up, like adding readonly to fields (note be careful of adding it to fields that are struct types, see [Mutating Readonly Structs](http://blogs.msdn.com/b/ericlippert/archive/2008/05/14/mutating-readonly-structs.aspx)) and eliminating some of the double casts.

**Pull request merged**